packages/terra-abstract-modal/src/AbstractModal.jsx

```diff
@@ -141,35 +124,34 @@ class AbstractModal extends React.Component {
     const mainDocumentElement = document.querySelector(this.props.rootSelector);
 
     if (mainDocumentElement) {
-      const inert = +mainDocumentElement.getAttribute('data-overlay-count');
+      const dataOverlayCount = +mainDocumentElement.getAttribute('data-abstract-modal-overlay-count');
 
-      if (inert === 1) {
-        mainDocumentElement.removeAttribute('data-overlay-count');
+      if (dataOverlayCount === 1) {
+        mainDocumentElement.removeAttribute('data-abstract-modal-overlay-count');
         mainDocumentElement.removeAttribute('inert');
-        // Ensures aria-hidden is properly cleaned up
-        setTimeout(() => mainDocumentElement.removeAttribute('aria-hidden'), 0);
-      } else if (inert && inert > 1) {
-        mainDocumentElement.setAttribute('data-overlay-count', `${inert - 1}`);
+      } else if (dataOverlayCount > 1) {
+        mainDocumentElement.setAttribute('data-abstract-modal-overlay-count', `${dataOverlayCount - 1}`);
       }
     }
 
-    setTimeout(() => {
-      if (this.state.modalTrigger && this.state.modalTrigger.focus) {
+    // Allows inert processing to finish or focus will not shift back
+    Promise.resolve().then(() => {
```

Comment:
Was the setTimeout not working? Seems like browsers could handle this differently based on whether the MutationObserver (or the polyfill) used with `inert` schedules micro- or macrotasks. Some fun reading about the differences in these implementations: https://jakearchibald.com/2015/tasks-microtasks-queues-and-schedules/

Reply:
The timeout was working, I was hoping to remove another timeout from the new-relic logs that get pushed out in our applications. I'll change it back